http_backend.go

```diff
@@ -148,7 +151,12 @@ func (h *httpBackend) Do(request *http.Request, bodySize int) (*Response, error)
 	if r != nil {
 		r.waitChan <- true
 		defer func(r *LimitRule) {
-			time.Sleep(r.Delay)
+			randomDelay := time.Duration(0)
+			rand.Seed(time.Now().UnixNano())
```

Comment:
Perhaps this line should be removed to be able to control the seed externally. It can be useful for testing and for reproducible runs. What do you think?

Reply:
I think that is a good idea, however seed can't be static, so I think we need to add something like this

` type LimitRule struct { SeedFunc func() int64 }`

what do you think?

Reply:
> I think that is a good idea, however seed can't be static

You're right.
What about calling `rand.Seed(time.Now().UnixNano())` from the package's `init()` function which runs only on startup?

Reply:
hmm, the seed has to be regenerate per request to achieve per-request random delays, generating a static seed in `init` might not work, or maybe I am misunderstanding?

We could also make all request having the same delay and make the randomness per process.

What do you think?

Reply:
Seed is used to initialize random generation. This little example demonstrates how it works: https://play.golang.com/p/4xks-DcCgv

More info: https://en.wikipedia.org/wiki/Random_seed

Reply:
yeah, you are right, for some reason I was thinking that I need to re seed to make each request random. Let me put the seed in init